data/api.py

```diff
@@ -2,16 +2,102 @@
 from rest_framework.response import Response
 from exceptions import DataServiceUnavailable
+from data.models import Workflow, WorkflowVersion, Job, JobInputFile, DDSJobInputFile, \
+    DDSEndpoint, DDSUserCredential, URLJobInputFile, JobError
+from data.serializers import WorkflowSerializer, WorkflowVersionSerializer, JobSerializer, \
+    DDSEndpointSerializer, DDSUserCredSerializer, JobInputFileSerializer, DDSJobInputFileSerializer, \
+    URLJobInputFileSerializer, JobErrorSerializer
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.decorators import detail_route
+from lando import LandoJob
 
 
 class ProjectsViewSet(viewsets.ViewSet):
-
     permission_classes = (permissions.IsAuthenticated,)
-
     def list(self, request):
         try:
             projects = get_user_projects(request.user)
             return Response(projects)
         except Exception as e:
             raise DataServiceUnavailable(e)
 
+
+class WorkflowsViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = Workflow.objects.all()
+    serializer_class = WorkflowSerializer
+
+
+class WorkflowVersionsViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = WorkflowVersion.objects.all()
+    serializer_class = WorkflowVersionSerializer
+
+
+class JobsViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = Job.objects.all()
+    serializer_class = JobSerializer
+
+    def perform_create(self, serializer):
+        self.save_with_user(serializer)
+
+    def perform_update(self, serializer):
+        self.save_with_user(serializer)
+
+    def save_with_user(self, serializer):
+        serializer.save(user=self.request.user)
+
+    @detail_route(methods=['post'])
+    def start(self, request, pk=None):
+        job = LandoJob(pk)
+        job.start()
+        return Response({'status': 'ok'})
+
+    @detail_route(methods=['post'])
+    def cancel(self, request, pk=None):
+        job = LandoJob(pk)
+        job.cancel()
+        return Response({'status': 'ok'})
+
+
+class DDSJobInputFileViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = DDSJobInputFile.objects.all().order_by('index')
+    serializer_class = DDSJobInputFileSerializer
+
+
+class JobInputFileViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = JobInputFile.objects.all()
+    serializer_class = JobInputFileSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filter_fields = ('job',)
+
+
+class URLJobInputFileViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = URLJobInputFile.objects.all().order_by('index')
+    serializer_class = URLJobInputFileSerializer
+
+
+class DDSEndpointViewSet(viewsets.ModelViewSet):
+    permission_classes = (permissions.IsAuthenticated,)
```

Comment:
I'll create a separate issue for it since it's beyond the scope of this initial functionality, but many of these ViewSets should have more restrictive permissions. IsAuthenticated allows any authenticated user to read and write the resources.